### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by nilab-khwaja**
- Create a new directory called logic
- implement click event handler 
- install big.js
- Import calculate.js in Calculator.js component

### CODE REVIEW

File: src/components/result.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ResultView.defaultProps = {
+  obj: {},
+};
+ResultView.propTypes = {
+  obj: PropTypes.shape({
+    total: PropTypes.string,
+    operation: PropTypes.string,
+    next: PropTypes.string,
+  }),
+};
```
Comment (Gambit142) [inline src/components/result.jsx]: - [ ] When I make some calculations, the results that are long are cut off and the entire results don't show. For example, in the video below: I did 46/3 and got this is what showed:

chrome-capture-2023-8-11

Kindly fix this and ensure that the long calculations show correctly. 👍 😎

File: src/components/result.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Gambit142) [inline src/components/result.jsx]: - [ ] Kindly remove the `eslint-disable` statement from this file. `Eslint` helps us to write clean, readable and industry-standard codes. Using `eslint-disable` statements defeats this purpose. Kindly remove it. 👍 💻

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @nilab-khwaja 👋, 

Good job so far!

## Highlights
- Set up linters correctly ✔️ 
- Use of logic directory ✔️ 
- Use of `useState` hook ✔️ 
- Passage of props between components ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by roshan-bajgain — APPROVED** [review]
Hi @nilab-khwaja, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- Gitflow is followed. ✔️
- Good job on changing requested ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**